dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumSessionScopeTest.kt

```diff
@@ -257,6 +260,69 @@ internal class RumSessionScopeTest {
         verifyZeroInteractions(mockWriter)
     }
 
+    @Test
+    fun `sends ApplicationStarted event on applicationDisplayed`(
+        @StringForgery(StringForgeryType.ALPHABETICAL) key: String,
+        @StringForgery(StringForgeryType.ALPHABETICAL) name: String
+    ) {
+        val childView: RumViewScope = mock()
+        val startViewEvent = RumRawEvent.StartView(key, name, emptyMap())
+
+        testedScope.onApplicationDisplayed(startViewEvent, childView, mockWriter)
+
+        argumentCaptor<RumRawEvent> {
+            verify(childView).handleEvent(capture(), same(mockWriter))
+
+            val event = firstValue as RumRawEvent.ApplicationStarted
+            assertThat(event.applicationStartupNanos).isEqualTo(Datadog.startupTimeNs)
+        }
+        verifyZeroInteractions(mockWriter)
+    }
+
+    @Test
+    fun `sends ApplicationStarted event on applicationDisplayed only once`(
+        @StringForgery(StringForgeryType.ALPHABETICAL) key: String,
+        @StringForgery(StringForgeryType.ALPHABETICAL) name: String
+    ) {
+        val childView: RumViewScope = mock()
+        val startViewEvent = RumRawEvent.StartView(key, name, emptyMap())
+
+        testedScope.onApplicationDisplayed(startViewEvent, childView, mockWriter)
+        testedScope.onApplicationDisplayed(startViewEvent, childView, mockWriter)
+
+        argumentCaptor<RumRawEvent> {
+            verify(childView).handleEvent(capture(), same(mockWriter))
```

Comment:
How does the "only once" assertion work here?

Reply:
It's part of our assertion framework. 
The `verify` methods has a second argument that checks the cardinality (so I could write `verify(myObject, times(2)).method()` or `verify(myObject, atLeastOnce()).method()`). By default the cardinality is set to exactly one